### PR TITLE
Implement Ruleset Selector component for profile overlay

### PR DESCRIPTION
**Pull request by EVAST9919**
Merge https://github.com/ppy/osu/pull/4964 first

Screenshot-528

### CODE REVIEW

**Comment by peppy** [comment]
There's already a similar control (`RulesetSelector`), could these be combined into a common base class? Also we don't use the term "game mode", so this should be called `ProfileRulesetSelector` or similar.

**Comment by EVAST9919** [comment]
Do we want to switch local game mode too?

**Comment by peppy** [comment]
No, I don't believe this should update the game-wide ruleset (the menu has control of this exclusively).

**Comment by peppy** [comment]
Are you intending to make this work (ie. update profile with correct ruleset)?

**Comment by EVAST9919** [comment]
> Are you intending to make this work (ie. update profile with correct ruleset)?

Yes. I already have some groundwork on this

**Comment by EVAST9919** [comment]
In this pr I just want to add a component, not the whole system. UserOverlay needs some refactoring.

**Comment by peppy** [comment]
This control should be in a disabled state or not yet added to the user profile (and only in a test case) if so. Else I wouldn't want to merge it as it is adding a control that doesn't actually work (which we are now avoiding)

**Comment by EVAST9919** [comment]
I'll remove that from the overlay, leaving it in testcases only

**Review by smoogipoo — DISMISSED** [review]

**Comment by smoogipoo** [comment]
@peppy quick pass then g2g

File: osu.Game/Overlays/Profile/Header/Components/ProfileRulesetTabItem.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                    Direction = FillDirection.Horizontal,
+                    Spacing = new Vector2(3, 0),
+                    Children = new Drawable[]
+                    {
+                        text = new OsuSpriteText
+                        {
+                            Origin = Anchor.Centre,
+                            Anchor = Anchor.Centre,
+                            Text = value.Name,
+                            Font = OsuFont.GetFont()
```
Comment (peppy) [inline osu.Game/Overlays/Profile/Header/Components/ProfileRulesetTabItem.cs]: is this doing anything?

**Comment by peppy** [comment]
Simplified state management. Fixes a potential issue (hover -> change accent colour would cause hover state to be lost).

**Review by peppy — APPROVED** [review]

**Pull request merged**